### PR TITLE
Map fields to docusign envelopes

### PR DESCRIPTION
**Pull request by tacataca**
Mapping fields from app to docusign.

- Add ability to manually send or revoke docusign envelope via child show page.
- Add ability to send a new docusign envelope if not possible to revoke the existing one.
- Adds ability to add note on email body instead of default message.
- Adds ability to specify the recipient(attendee) of the email. Defaults to either parent.
- Determines which template to use based on child grade and presence of misc medical info.
- Recipient email is overwritten to test email unless production.

### CODE REVIEW

File: app/assets/javascripts/child/docusign.coffee [hunk 1]
```diff
@@ -0,0 +1,16 @@
+$ ->
+  return unless $('.show.children').length > 0
+
+  $('#void_button').click ->
+    confirm "
+    Only pending envelopes are voidable via this page.
```
Comment (sheldond) [inline app/assets/javascripts/child/docusign.coffee]: The meaning of this first sentence wasn't immediately clear to me, do you think it is necessary? I just wanted to make sure that any user is not confused in any way.

File: app/models/attendee.rb [hunk 2]
```diff
@@ -56,6 +56,18 @@ def exempt?
     conference_status == 'Exempt'
   end
 
+  def main_ministry
+    ministry.ancestors[1]
+  end
+
+  def cohort
```
Comment (sheldond) [inline app/models/attendee.rb]: Should this method `return nil if !campus_ministry_member?`

**Comment by tacataca** [comment]
Im not too sure on what to do for testing.
Any suggestions on what to focus on?

**Comment by sheldond** [comment]
A simple test for each of the new controller actions would go a long way I think in helping prevent any regressions (the jobs themselves are already tested so we wouldn't have to test all of that again).

There are some new methods in `Attendee` and in `Childcare::SendDocusignEnvelope`, some of them seem very simple and potentially not worth testing, but others have some logic, it might be worth considering simple unit tests for some of these.

**Pull request merged**